packages/types/src/cart/service.ts

```diff
@@ -1,4 +1,48 @@
-import { IModuleService } from "../modules-sdk";
+import { FindConfig } from "../common"
+import { IModuleService } from "../modules-sdk"
+import { Context } from "../shared-context"
+import { CartDTO, FilterableCartProps } from "./common"
+import { AddLineItemsDTO, CreateCartDTO, UpdateCartDTO, UpdateLineItemsDTO } from "./mutations"
 
 export interface ICartModuleService extends IModuleService {
+  retrieve(
+    cartId: string,
+    config?: FindConfig<CartDTO>,
+    sharedContext?: Context
+  ): Promise<CartDTO>
+
+  list(
+    filters?: FilterableCartProps,
+    config?: FindConfig<CartDTO>,
+    sharedContext?: Context
+  ): Promise<CartDTO[]>
+
+  listAndCount(
+    filters?: FilterableCartProps,
+    config?: FindConfig<CartDTO>,
+    sharedContext?: Context
+  ): Promise<[CartDTO[], number]>
+
+  create(data: CreateCartDTO[], sharedContext?: Context): Promise<CartDTO[]>
+
+  update(data: UpdateCartDTO[], sharedContext?: Context): Promise<CartDTO[]>
+
+  delete(productIds: string[], sharedContext?: Context): Promise<void>
+
+  addLineItems(data: AddLineItemsDTO, sharedContext?: Context): Promise<CartDTO>
+  addLineItems(
+    data: AddLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartDTO[]>
+
+  updateLineItems(
+    data: UpdateLineItemsDTO,
+    sharedContext?: Context
+  ): Promise<CartDTO>
+  updateLineItems(
+    data: UpdateLineItemsDTO[],
+    sharedContext?: Context
+  ): Promise<CartDTO[]>
```

Comment:
question: Wouldn't it make sense for those methods to return the item, and then you can refresh/retrieve your cart with the relations etc that you need?

Reply:
Yes, you might be right. I haven't started implementing these methods and have actually commented them out in other follow-up PRs. Will do the same in this one, and then we can discuss when I get to the point of building them. 

But yes, I think it would make sense to return the cart.

Ok with you?

Reply:
We can, but I thought those should return the items, like the address return the address and cart return cart. But I am fine to return the cart, it is just that in general the user will still re fetch the cart to get the proper relations unless we load them all

Reply:
Sorry, I meant the line items. I agreed with your point 😅

Reply:
ok then :D no problemn